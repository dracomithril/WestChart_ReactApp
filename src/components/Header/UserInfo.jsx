/**
 * Created by Gryzli on 07.04.2017.
 */
import React from "react";
import PropTypes from 'prop-types';
import {Button, Image, Well} from "react-bootstrap";

const { writeUserData } = require('./../../utils');
const action_types = require('./../../reducers/action_types');

export default class UserInfo extends React.Component {
  /*istanbul ignore next*/
  componentDidMount() {
    console.log('component UserInfo did mount');
    const { store } = this.context;
    let { user, sp_user } = store.getState();
    if (user.id && sp_user.id) {
      writeUserData(user, sp_user)
    }

  }

  logoutUser = () => {
    const { store } = this.context;
    store.dispatch({ type: action_types.SIGN_OUT_USER });
    sessionStorage.removeItem('fb_user');
    sessionStorage.removeItem('sp_user');
  };

  /*istanbul ignore next*/
  componentWillUnmount() {
    console.log('component UserInfo unmounted');
  }

  render() {
    const { store } = this.context;
    let { user, sp_user } = store.getState();
    return (
      <Well bsClass="logged">
        <div>
          <Image src={user.picture_url} circle/>
          <span>{`Hi, ${user.first_name}`}</span><br/>
          <span>{`it's nice to see you again.`}</span><br/>
          {sp_user.id !== undefined &&
          <div className="spotify">
            <i className="fa fa-spotify"/>
            <span>{sp_user.id}</span>
          </div>}
          <div style={{ textAlign: "center" }}>
            <Button bsStyle="warning" onClick={this.logoutUser}>Sign out</Button>
          </div>
        </div>
      </Well>);
  }
}
UserInfo.contextTypes = {
  store: PropTypes.object
};

UserInfo.propTypes = {};