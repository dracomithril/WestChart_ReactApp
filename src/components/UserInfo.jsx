/**
 * Created by Gryzli on 07.04.2017.
 */
import React from "react";
import PropTypes from 'prop-types';
// import Spotify from 'spotify-web-api-js';
import {Image, Well} from "react-bootstrap";
export default class UserInfo extends React.Component {
    componentWillUnmount() {
        console.log('component UserInfo unmounted');
    }

    componentDidMount() {
         // const {store} = this.context;
        console.log('component UserInfo did mount');
    }

    render() {
        const {store} = this.context;
        let {user, sp_user} = store.getState();
        return (
            <Well bsClass="logged">
                <div>
                    <Image src={user.picture_url} circle/>
                    <span>{`Hi, ${user.first_name}`}</span><br/>
                    <span>{`it's nice to see you again.`}</span><br/>
                    {sp_user.id !== undefined &&
                    <div className="spotify">
                        <i className="fa fa-spotify"/>
                        <span >{sp_user.id}</span>
                    </div>}
                </div>
            </Well>);
    }
}
UserInfo.contextTypes = {
    store: PropTypes.object
};

UserInfo.propTypes = {};