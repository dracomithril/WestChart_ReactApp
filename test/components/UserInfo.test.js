/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import UserInfo from '../../src/components/Header/UserInfo';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('./../data/initial_state.json');

const mockStore = configureMockStore([]);

describe('<UserInfo/>', () => {
    it('renders without crashing', () => {

        const store = mockStore(initial_state);
        const wrapper = shallow(
            <UserInfo />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
