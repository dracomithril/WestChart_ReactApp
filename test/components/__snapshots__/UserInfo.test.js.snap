// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo/> renders without crashing 1`] = `
<Well
  bsClass="logged"
>
  <div>
    <Image
      bsClass="img"
      circle={true}
      responsive={false}
      rounded={false}
      thumbnail={false}
    />
    <span>
      Hi, undefined
    </span>
    <br />
    <span>
      it's nice to see you again.
    </span>
    <br />
  </div>
</Well>
`;
